include/circt/Dialect/Calyx/CalyxLoweringUtils.h

```diff
@@ -424,6 +424,15 @@ class ComponentLoweringStateInterface {
     Block *body = component.getBodyBlock();
     builder.setInsertionPoint(body, body->begin());
     auto name = TLibraryOp::getOperationName().split(".").second;
+    if constexpr (TLibraryOp::template hasTrait<FloatingPoint>()) {
+      auto standard = TLibraryOp::getFloatingPointStandard();
+      if (standard == FloatingPointStandard::IEEE754) {
+        assert(name.consume_front("ieee754.") &&
+               "IEEE754 type operation's name must begin with 'ieee754'");
+        std::string modified = llvm::join_items("", "std_", name, "FN");
```

Comment:
Do we need the empty string in the front?

Reply:
I believe so, otherwise it thinks `std_` is the separator

Reply:
Ah yes sorry.

Reply:
sure, good idea!